app/uk/gov/hmrc/helptosave/models/PayePersonalDetails.scala

```diff
@@ -102,30 +102,29 @@ object PayePersonalDetails {
     readSeq(json, "addresses", "2").orElse(readSeq(json, "addresses", "1")) //1–Residential Address, 2–Correspondence Address
       .fold[JsResult[Address]](
         JsError("No Address found in the DES response")
-      )(v ⇒ {
-          val line1 = (v \ "line1").as[String]
-          val line2 = (v \ "line2").as[String]
-          val line3 = (v \ "line3").asOpt[String]
-          val line4 = (v \ "line4").asOpt[String]
-          val line5 = (v \ "line5").asOpt[String]
-          val postcode = (v \ "postcode").as[String]
-          val countryCode = (v \ "countryCode").asOpt[Int]
-
-          JsSuccess(Address(line1, line2, line3, line4, line5, postcode, countryCode.flatMap(countryCodes.get).map(_.take(2))))
-        }
+      )(v ⇒
+          for {
+            line1 ← (v \ "line1").validate[String]
+            line2 ← (v \ "line2").validate[String]
+            line3 ← (v \ "line3").validateOpt[String]
+            line4 ← (v \ "line4").validateOpt[String]
+            line5 ← (v \ "line5").validateOpt[String]
+            postcode ← (v \ "postcode").validate[String]
```

Comment:
would this postcode be evaluated if the options line3, line4, or line5 are not present ?

Reply:
yes, the `flatMap` is being done on  `JsResult` rather than on `Option`

Reply:
ah ok , no issues then